### DISCUSSION TITLE
IamForce2 - shift+button mapping?

### DISCUSSION THREAD

**Discussion by legsmechanical** [TheKikGen/MPCLiveXplore-libs#11] [Q&A]
Howdy! I'm curious whether it's possible to edit the controller templates to map a shift combo directly to a single button on the mpc.  Ideally, I'd like to be able to hold a single button that puts the pads into the track select/mute/solo state AND engages shift so that the launch quantization/metronome/tc/etc functions are immediately available (rather than holding the current track mode button and also shift to enter that mode).  Thanks!

**Comment by TheKikGen** [comment]
Hi.  Yes possible.  What is your controller ?

**Comment by legsmechanical** [comment]
Awesome. I use an LPX but have recently started getting in well with the
standard mapping for Mpc live 2 (which I’ve already edited and recompiled
myself to some degree.)

**Reply by TheKikGen** [reply]
It is quite easy to modify the code, because the mapping it self is clearly separated.  Have a look at the APC Key 25 plugin source.  I use intensively the controller SHIFT and STOP ALL CLIPS keys for combinations.  
https://github.com/TheKikGen/MPCLiveXplore-libs/blob/df1be83ec209a1e8be63759bb05fa217b4f0d0ae/src/tmm-plugins/Iamforce-APCKEY25MK2.h#L410

**Comment by legsmechanical** [comment]
Will do. Thanks!

(And just to be clear, I’m looking to have one Dutton on the controller
send the shift+another function simultaneously. )
